L1Trigger/TrackFindingTracklet/src/TrackletProcessorDisplaced.cc

```diff
@@ -233,461 +158,156 @@ void TrackletProcessorDisplaced::addInput(MemoryBase* memory, string input) {
+  for (unsigned int midmem = 0; midmem < middleallstubs_.size(); midmem++) {
+    for (unsigned int i = 0; i < middleallstubs_[midmem]->nStubs(); i++) {
+      const Stub* midallstub = middleallstubs_[midmem]->getStub(i);
 
       if (settings_.debugTracklet()) {
-        edm::LogVerbatim("Tracklet") << "In " << getName() << " have first stub\n";
+        edm::LogVerbatim("Tracklet") << "In " << getName() << " have middle stub";
       }
 
-      int inner = 0;
-      bool negdisk = (firstallstub->disk().value() < 0);
-      int indexz = (((1 << (firstallstub->z().nbits() - 1)) + firstallstub->z().value()) >>
-                    (firstallstub->z().nbits() - nbitszfinebintable_));
+      // get r/z index of the middle stub
+      int indexz = (((1 << (midallstub->z().nbits() - 1)) + midallstub->z().value()) >>
+                    (midallstub->z().nbits() - nbitszfinebintable_));
       int indexr = -1;
-      if (layerdisk_ > (N_LAYER - 1)) {
-        if (negdisk) {
+      bool negdisk = (midallstub->disk().value() < 0);  // check if disk in negative z region
+      if (layerdisk1_ >= LayerDisk::D1) {               // if a disk
+        if (negdisk)
           indexz = (1 << nbitszfinebintable_) - indexz;
+        indexr = midallstub->r().value();
+        if (midallstub->isPSmodule()) {
+          indexr = midallstub->r().value() >> (midallstub->r().nbits() - nbitsrfinebintable_);
         }
-        indexr = firstallstub->r().value();
-        if (firstallstub->isPSmodule()) {
-          indexr = firstallstub->r().value() >> (firstallstub->r().nbits() - nbitsrfinebintable_);
-        }
-      } else {
-        //Take the top nbitsfinebintable_ bits of the z coordinate. The & is to handle the negative z values.
-        indexr = (((1 << (firstallstub->r().nbits() - 1)) + firstallstub->r().value()) >>
-                  (firstallstub->r().nbits() - nbitsrfinebintable_));
+      } else {  // else a layer
+        indexr = (((1 << (midallstub->r().nbits() - 1)) + midallstub->r().value()) >>
+                  (midallstub->r().nbits() - nbitsrfinebintable_));
       }
 
       assert(indexz >= 0);
       assert(indexr >= 0);
       assert(indexz < (1 << nbitszfinebintable_));
       assert(indexr < (1 << nbitsrfinebintable_));
 
-      // int melut = meTable_.lookup((indexz << nbitsrfinebintable_) + indexr);
-      // assert(melut >= 0);
-
-      unsigned int lutwidth = settings_.lutwidthtab(inner, iSeed_);
-      if (settings_.extended()) {
-        lutwidth = settings_.lutwidthtabextended(inner, iSeed_);
-      }
-
-      int lutval = -999;
-
-      if (iSeed_ < Seed::L1D1 || iSeed_ > Seed::L2D1) {
+      // create lookupbits that define projections from middle stub
+      unsigned int lutwidth = settings_.lutwidthtabextended(0, iSeed_);
+      int lutval = -1;
+      const auto& lutshift = innerTable_.nbits();
+      if (iSeed_ == Seed::L2L3L4 || iSeed_ == Seed::L4L5L6 || iSeed_ == Seed::D1D2L2 || iSeed_ == Seed::L2L3D1) {
```

Comment:
@cgsavard I believe this conditional is redundant, i.e., `iSeed_` is always one of these four values in the TPD, so it can be removed.